core/src/main/java/org/infinispan/marshall/exts/ArrayExternalizer.java

```diff
@@ -0,0 +1,114 @@
+package org.infinispan.marshall.exts;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.infinispan.commons.io.UnsignedNumeric;
+import org.infinispan.commons.marshall.AbstractExternalizer;
+import org.infinispan.commons.marshall.AdvancedExternalizer;
+import org.infinispan.commons.marshall.MarshallUtil;
+import org.infinispan.commons.util.Util;
+import org.infinispan.marshall.core.Ids;
+import org.jboss.marshalling.util.IdentityIntMap;
+
+/**
+ * Externalizers for diverse array types.
+ *
+ * @author Galder Zamarreño
+ * @since 6.0
+ */
+public class ArrayExternalizer extends AbstractExternalizer<Object> {
```

Comment:
Do we still need these array externalizers in the internal marshaller? IMO even `MarshallUtil.marshallArray` is too generic for our internal objects, as most of the time we should know exactly what kind of objects we are trying to marshall/unmarshall.